### PR TITLE
Enforce use_raa even if user is raa through select_raa

### PR DESCRIPTION
**Pull request by pablothedude**
There was a bug which allowed RAA's to execute actions if
the user was made RAA through select_raa. There however still
needed to be validated if a user could act as RAA on a
certain institution by also enforcing use_raa.

https://www.pivotaltracker.com/story/show/170487212

### CODE REVIEW

File: src/Surfnet/StepupMiddleware/CommandHandlingBundle/Command/RaExecutable.php [hunk 1]
```diff
@@ -23,4 +23,9 @@
  */
 interface RaExecutable
 {
+    /**
+     * Returns ra institution for authorization purposes
+     * @return string|null
+     */
+    public function getRaInstitution();
```
Comment (MKodde) [inline src/Surfnet/StepupMiddleware/CommandHandlingBundle/Command/RaExecutable.php]: +1 for adding this, please update the phpdoc comment of the interface itself. It mentions this is a marker interface, but it not longer is. Same goes for the SelfServiceExecutable.

File: src/Surfnet/StepupMiddleware/ApiBundle/Controller/CommandController.php [hunk 2]
```diff
@@ -182,4 +192,94 @@ private function assertCommandMayBeProcessedOnBehalfOf($institution, $actorId)
             $actorId
         ));
     }
+
+    /**
+     * @param string $actorId
+     * @param Command $command
+     */
+    private function assertSelfServiceCommandMayBeExecutedOnBehalfOf($actorId, Command $command)
```
[the remark this answers is not in the document]
Comment (pablothedude) [reply src/Surfnet/StepupMiddleware/ApiBundle/Controller/CommandController.php]: Done

File: src/Surfnet/StepupMiddleware/ApiBundle/Controller/CommandController.php [hunk 3]
```diff
@@ -182,4 +192,94 @@ private function assertCommandMayBeProcessedOnBehalfOf($institution, $actorId)
...
+                ));
+            }
+        }
+    }
+
+    /**
+     * @param string $actorId
+     * @return bool
+     */
+    private function isSraa($actorId)
```
[the remark this answers is not in the document]
Comment (pablothedude) [reply src/Surfnet/StepupMiddleware/ApiBundle/Controller/CommandController.php]: Done

File: src/Surfnet/StepupMiddleware/ApiBundle/Controller/CommandController.php [hunk 4]
```diff
@@ -100,7 +112,15 @@ public function handleAction(Command $command, Metadata $metadata, Request $requ
...
-            $this->assertCommandMayBeProcessedOnBehalfOf($institution, $metadata->actorId);
+            $this->assertInstitutionIsWhitelisted($institution, $metadata->actorId);
+
+
+            $logger->notice('Ensuring that the actor is allowed to execute a command based on the fine grained authorization configuration');
+
+            $this->assertSelfServiceCommandMayBeExecutedOnBehalfOf($metadata->actorId, $command);
+            $this->assertRaCommandMayBeExecutedOnBehalfOf($metadata->actorId, $metadata->actorInstitution, $command);
+
+            $logger->notice('Command authorization succeeded');
```
[the remark this answers is not in the document]
Comment (pablothedude) [reply src/Surfnet/StepupMiddleware/ApiBundle/Controller/CommandController.php]: Removed

File: src/Surfnet/StepupMiddleware/ApiBundle/Controller/CommandController.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (MKodde) [reply src/Surfnet/StepupMiddleware/ApiBundle/Controller/CommandController.php]: Thanks for adding the new service and covering it with tests!

**Pull request merged**